test/unit/specs/validators/email.spec.js

```diff
@@ -0,0 +1,48 @@
+import email from 'src/validators/email'
+
+describe('email validator', () => {
+  it('should not validate empty string', () => {
+    expect(email('')).to.be.false
+  })
+
+  it('should not validate numbers', () => {
+    expect(email('12345')).to.be.false
+  })
+
+  it('should not validate strings', () => {
+    expect(email('asdf12345')).to.be.false
+  })
+
+  it('should not validate space', () => {
+    expect(email(' ')).to.be.false
+  })
+
+  it('should not validate incomplete addresses', () => {
+    expect(email('someone@')).to.be.false
+    expect(email('someone@gmail')).to.be.false
+    expect(email('someone@gmail.')).to.be.false
+    expect(email('someone@gmail.c')).to.be.false
+    expect(email('gmail.com')).to.be.false
+    expect(email('@gmail.com')).to.be.false
+  })
+
+  it('should not validate addresses that contain unsupported characters', () => {
+    expect(email('someone@g~mail.com')).to.be.false
+    expect(email('some!one@gmail.com')).to.be.false
```

Comment:
Exclamation mark is perfectly valid character for email address.

Validating emails in general is quite hard, as RFC has many details that people often forget about. Consider making your regex a bit more relaxed. It's always better to allow some false positives than to be too strict in validation. Proper email validation would involve querying the DNS for MX record, and this requires a backend service, so we can't really expect our validator to be perfect anyway.